tests/integration/java/com/github/bazel_contrib/target_determinator/integration/BazelDifferIntegrationTest.java

```diff
@@ -53,7 +56,19 @@ private static void runProcess(Path workingDirectory, String... argv)
     // Do not clean the environment so we can inherit variables passed e.g. via --test_env.
     // Useful for CC (needed by bazel).
     processBuilder.environment().put("HOME", System.getProperty("user.home"));
-    processBuilder.environment().put("PATH", System.getenv("PATH"));
+
+    // There is a bug in `bazel-differ` where in `internal/bazel.go` they default to always using `bazel`
+    // rather than the the actual bazel instance we've requested. This is normally fine (just fine) but
+    // when we execute our own tests with `bazelisk`, the `PATH` has the version of `bazel` requested in
+    // our _own_ `.bazelversion` file at the head of the `PATH` (pulled from `bazelisk`'s own cache). To
+    // avoid this, we're going to remove items from the `PATH` that might be from bazelisk using a very
+    // sophisticated heuristic.
+    String cacheDir = getUserCacheDirectory();
+    String amendedPath = Arrays.stream(System.getenv("PATH").split(File.pathSeparator))
+            .filter(item -> !item.contains(cacheDir))
```

Comment:
Bazel-differ looks like it's not been updated in a while, so I don't want to be a pain but this filter is still a bit too indiscriminate IMO. Could at least append the bazelisk subdirectory? Yes it's probably an implementation detail of bazelisk but it feels like the surprise effect would be lower than if something else under the user cache directory gets removed and caused grief to users.

Reply:
I think we're the only people who run these tests, and as long as it works in CI, then we've done sufficiently well, I think. I can append some more, and hope that bazelisk uses the same in-cache path for every platform.